java/format/src/main/java/org/apache/arrow/flatbuf/Feature.java

```diff
@@ -37,26 +37,23 @@
  * to facilitate exchanging and comparing bitmaps for supported
  * features.
  */
+@SuppressWarnings("unused")
 public final class Feature {
   private Feature() { }
   /**
    * Needed to make flatbuffers happy.
    */
-  public static final long UNUSED = 0;
+  public static final long UNUSED = 0L;
   /**
    * The stream makes use of multiple full dictionaries with the
    * same ID and assumes clients implement dictionary replacement
    * correctly.
    */
-  public static final long DICTIONARY_REPLACEMENT = 1;
+  public static final long DICTIONARY_REPLACEMENT = 1L;
   /**
    * The stream makes use of compressed bodies as described
    * in Message.fbs.
    */
-  public static final long COMPRESSED_BODY = 2;
-
-  public static final String[] names = { "UNUSED", "DICTIONARY_REPLACEMENT", "COMPRESSED_BODY", };
-
-  public static String name(int e) { return names[e]; }
```

Comment:
Hmm, weird that it doesn't generate this anymore. We should probably be conservative and note the breaking change, even if I don't think anyone would have directly touched the flatbufs.

Reply:
I've also filed https://github.com/google/flatbuffers/issues/8179 in the meantime.

Reply:
Okay, I verified that this is done by design for java flatbuffer generation (see issue above). Since Enum values actually reflect the `long` type when specified now, we can't create an array and try to index into it with a long value or we'll get a compiler error.

Reply:
I think we can't change that without breaking other things.

Reply:
That's fair. I wouldn't want to modify the format itself for this tbh.